common/gpu/intel/default.nix

```diff
@@ -1,23 +1,42 @@
-{ config, lib, pkgs, ... }:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
 
 {
-  options.hardware.intelgpu.loadInInitrd = lib.mkEnableOption (lib.mdDoc
-    "loading `i195` kernelModule at stage 1. (Add `i915` to `boot.initrd.kernelModules`)"
-  ) // {
-    default = true;
-  };
+  options.hardware.intelgpu.loadInInitrd =
+    lib.mkEnableOption (
+      lib.mdDoc "loading `i195` kernelModule at stage 1. (Add `i915` to `boot.initrd.kernelModules`)"
+    )
+    // {
+      default = true;
+    };
+
+  options.hardware.intelgpu.loadXeInInitrd =
+    lib.mkEnableOption (
+      lib.mdDoc "loading `xe` kernelModule at stage 1. (Add `xe` to `boot.initrd.kernelModules`)"
+    )
+    // {
+      default = false;
+    };
 
   config = lib.mkMerge [
-    (lib.mkIf config.hardware.intelgpu.loadInInitrd { 
-      boot.initrd.kernelModules = [ "i915" ]; 
-    })
+    (lib.mkIf config.hardware.intelgpu.loadInInitrd { boot.initrd.kernelModules = [ "i915" ]; })
+    (lib.mkIf config.hardware.intelgpu.loadXeInInitrd { boot.initrd.kernelModules = [ "xe" ]; })
```

Comment:
so xe is for newer devices only, right?

Reply:
ya it was introduced in the tiger lake chipset, so only it, alder lake, and raptor lake will have it afaik. I don't have an alder or raptor lake system to test so didn't bother adding those.